### DISCUSSION TITLE
getting error while running npm run build for the production build

### DISCUSSION THREAD

**Discussion by rajcoutlloot** [vercel/next.js#14125] [Help]
Hi ,
     while running npm run build for the production build of the nextjs app
    I am getting this error

    Compiled successfully.

Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`
Read more: https://err.sh/next.js/opt-out-auto-static-optimization

Automatically optimizing pages ..
Error occurred prerendering page "/404". Read more: https://err.sh/next.js/prerender-error:
TypeError: Cannot read property 'encrypted' of undefined

Error occurred prerendering page "/". Read more: https://err.sh/next.js/prerender-error:
TypeError: Cannot read property 'encrypted' of undefined

Error occurred prerendering page "/404.html". Read more: https://err.sh/next.js/prerender-error:
TypeError: Cannot read property 'encrypted' of undefined

> Build error occurred
Error: Export encountered errors
    at _default (/home/osis/raj_server/mobile-web/node_modules/next/dist/export/index.js:19:1164)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async build (/home/osis/raj_server/mobile-web/node_modules/next/dist/build/index.js:36:218)
Automatically optimizing pages .npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mobile-web@0.1.0 build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mobile-web@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/osis/.npm/_logs/2020-06-12T12_20_18_063Z-debug.log

**Comment by portenez** [comment]
nextjs runs an optimization step that will "run" your during the optimization step. what's probably happening, is that something in your env is `undefined` and the build fails. 

It's happening to me too. I was able to fix it, but I wonder if there's a way to make next not do this `run` when optimizing.

If you're reading something from your env... I think you can provide it as env variables or in an `env` file.

**Comment by rpunjwnaixcute** [comment]
I am also getting this error. How to fix this?
